### DISCUSSION TITLE
Set Custom donation levels Programmatically

### DISCUSSION THREAD

**Discussion by 007amrit** [impress-org/givewp#7152] [Developer Q&A]
Hello 

I am referred here by Premium support 

I am trying to create donation forms from front end using gravity forms and want to be able to set custom donation levels 

I am doing this 

```
add_action('pending_to_publish', 'oj_publish_campaign',10 ,1);

function oj_publish_campaign($post) {
    global $wpdb;
 
    $pid = $post->ID;

    $t = get_post_meta( $pid, '_give_donation_levels', true );    

    if(is_array($t)){
        $t[0]["_give_amount"] = 25.000000;
        $t[1]["_give_amount"] = 650.000000;              
        $t[2]["_give_amount"] = 1000.00000;
        $t[3]["_give_amount"] = 2500.00000;
        $t[4]["_give_amount"] = 6500.00000;
       
        $t[0]["_give_default"] = '';
        $t[1]["_give_default"] = 'default';
        $t[2]["_give_default"] = '';
        $t[3]["_give_default"] = '';
        $t[4]["_give_default"] = '';
    }

    update_post_meta( $pid, '_give_donation_levels', $t );

}
```

This should work but sadly its not working.

but once I put a published form/post back to pending status
and then republishes it then it works

Also I noticed that if use gravity form to auto publish directly then there are no donation level at all

so is there a way to address that too? Please help

**Comment by rednevednav** [comment]
You can wrap in a function in a dropin plugin....something like:

...stuff before
`    // Ensure the '_give_donation_levels' meta key is populated
    $default_donation_levels = [
        [
            '_give_id' => ['level_id' => '0'],
            '_give_amount' => '10.000000',
            '_give_text' => '',
            '_give_recurring' => 'yes',
            '_give_period_interval' => '1',
            '_give_period' => 'month',
            '_give_times' => '0',
        ],
        [
            '_give_id' => ['level_id' => '1'],
            '_give_amount' => '25.000000',
            '_give_text' => '',
            '_give_recurring' => 'yes',
            '_give_period_interval' => '1',
            '_give_period' => 'month',
            '_give_times' => '0',
        ],
        [
            '_give_id' => ['level_id' => '2'],
            '_give_amount' => '50.000000',
            '_give_text' => '',
            '_give_recurring' => 'yes',
            '_give_period_interval' => '1',
            '_give_period' => 'month',
            '_give_times' => '0',
        ],
        [
            '_give_id' => ['level_id' => '3'],
            '_give_amount' => '100.000000',
            '_give_text' => '',
            '_give_default' => 'default',
            '_give_recurring' => 'yes',
            '_give_period_interval' => '1',
            '_give_period' => 'month',
            '_give_times' => '0',
        ],
        [
            '_give_id' => ['level_id' => '5'],
            '_give_amount' => '250.000000',
            '_give_text' => '',
            '_give_recurring' => 'yes',
            '_give_period_interval' => '1',
            '_give_period' => 'month',
            '_give_times' => '0',
        ],
    ];

    update_post_meta($post_id, '_give_donation_levels', $default_donation_levels);
`
...stuff after
